doc/developer/design/20251215_jwt_authentication.md

```diff
@@ -0,0 +1,146 @@
+# JWT Authentication
+
+- Associated: (Insert list of associated epics, issues, or PRs)
+
+PRD: https://www.notion.so/materialize/SSO-for-Self-Managed-2a613f48d37b806f9cb2d06914454d15
+
+
+## Problem
+
+Our goal is to enable single sign on (SSO) for our self managed product
+
+## Success Criteria
+
+- Creating a user & adding roles: An admin gives a user access to Materialize
+- The user should be disabled from logging in when a user is de-provisioned. However, the database level role should still exist.
+- The end user is able to create a token to connect to materialize via psql / postgres clients
+- The end user is able to visit the Materialize console, and sign in with their IdP
+
+## Configuration
+
+JWT authentication can be enabled by specifying the following Materialize CRD:
+
+```yaml
+apiVersion: materialize.cloud/v1alpha1
+kind: Materialize
+metadata:
+  ...
+spec:
+  ...
+  authenticatorKind: Jwt
+  jwtAuthenticationSettings:
+  # Must match the OIDC client ID. Required.
+    audience: 060a4f3d-1cac-46e4-b5a5-6b9c66cd9431
+    # The claim that represents the user's name in Materialize.
+    # Usually either "sub" or "email". Required.
+    authenticationClaim: email
+    # The key for the `groups` claim. Optional and defaults to "groups"
+    groupClaim: groups
+    # The expected issuer URL, must correspond to the `iss` field of the JWT.
+    # Implicitly fetch JWKS from
+    # https://{ domain }/.well-known/openid-configuration and allow override
+    # if we need to. Required.
+    issuer: https://dev-123456.okta.com/oauth2/default
+    # Where Materialize will request tokens from the IdP using the refresh token
+    # if it exists. Optional.
+    tokenEndpoint: https://dev-123456.okta.com/oauth2/default/v1/token
```

Comment:
Is this actually standardized? How do we know what kind of request to make? If this is specifically an OIDC thing we should make that clear.